noxfile.py

```diff
@@ -0,0 +1,267 @@
+    ).strip()
+    session.run(
+        "pyre",
+        "--source-directory",
+        "src",
+        "--search-path",
+        purelib,
+        "--search-path",
+        platlib,
+        "--search-path",
+        "stubs/",
+        *additional_libraries,
+    )
+
+
+def _repl_version(session: nox.Session, new_version: str):
+    with open("pyproject.toml", "r+") as fp:
+        lines = tuple(fp)
+        fp.seek(0)
+        for line in lines:
+            if line.startswith("version = "):
+                line = f'version = "{new_version}"\n'
+            fp.write(line)
+        fp.truncate()
+
+
+def check_no_modifications(session: nox.Session) -> None:
+    modified = session.run(
+        "git",
+        "status",
+        "--porcelain=v1",
+        "--untracked=normal",
+        external=True,
+        silent=True,
+    )
+    if modified:
+        session.error(
+            "There are modified or untracked files. "
+            "Commit, restore, or remove them before running this"
+        )
+
+
+@nox.session
+def bump(session: nox.Session):
+    check_no_modifications(session)
+    if len(session.posargs) not in (1, 2):
+        session.error(
+            "Must specify 1-2 positional arguments: nox -e bump -- <version> "
+            "[ <release_summary_message> ]."
+            " If release_summary_message has not been specified, "
+            "a file changelogs/fragments/<version>.yml must exist"
+        )
+    version = session.posargs[0]
+    fragment_file = Path(f"changelogs/fragments/{version}.yml")
+    if len(session.posargs) == 1:
+        if not fragment_file.is_file():
+            session.error(
+                f"Either {fragment_file} must already exist, "
+                "or two positional arguments must be provided."
+            )
+    install(session, "antsibull-changelog", "hatch", "tomli ; python_version < '3.11'")
+    _repl_version(session, version)
+    if len(session.posargs) > 1:
+        fragment = session.run(
+            "python",
+            "-c",
+            "import yaml ; "
+            f"print(yaml.dump(dict(release_summary={repr(session.posargs[1])})))",
+            silent=True,
+        )
+        with open(fragment_file, "w") as fp:
+            print(fragment, file=fp)
+        session.run("git", "add", "pyproject.toml", fragment_file, external=True)
+        session.run("git", "commit", "-m", f"Prepare {version}.", external=True)
+    session.run("antsibull-changelog", "release")
+    session.run(
+        "git",
+        "add",
+        "CHANGELOG.rst",
+        "changelogs/changelog.yaml",
+        "changelogs/fragments/",
+        external=True,
+    )
+    install(session, ".")  # Smoke test
+    session.run("git", "commit", "-m", f"Release {version}.", external=True)
+    session.run(
+        "git",
+        "tag",
+        "-a",
+        "-m",
+        f"antsibull-docs {version}",
+        "--edit",
+        version,
+        external=True,
+    )
+    session.run("hatch", "build")
```

Comment:
This will clean existing artifacts before building. I think we should start using this in the other projects, also. If there's other artifacts in the `dist` directory (e.g. a test build for a `.post0` version that shouldn't be released), `hatch publish` will publish it.

```suggestion
    session.run("hatch", "build", "--clean")
```

Reply:
Yeah, I noticed that it tries to upload everything it finds in `dist/`. Cleaning up is thus :+1:.